components/p3a/constellation_helper_unittest.cc

```diff
@@ -0,0 +1,181 @@
+#include "content/public/test/browser_task_environment.h"
+#include "services/network/public/cpp/weak_wrapper_shared_url_loader_factory.h"
+#include "services/network/test/test_url_loader_factory.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace p3a {
+
+namespace {
+
+constexpr uint8_t kTestEpoch = 5;
+constexpr char kTestNextEpochTime[] = "2086-06-24T18:00:00Z";
+constexpr char kTestHistogramName[] = "Brave.Test.Histogram";
+constexpr char kTestHost[] = "https://localhost:8443";
+
+}  // namespace
+
+class P3AConstellationHelperTest : public testing::Test {
+ public:
+  P3AConstellationHelperTest()
+      : task_environment_(base::test::TaskEnvironment::TimeSource::MOCK_TIME),
+        shared_url_loader_factory(
+            base::MakeRefCounted<network::WeakWrapperSharedURLLoaderFactory>(
+                &url_loader_factory)) {}
+
+ protected:
+  void SetUp() override {
+    p3a_config.disable_star_attestation = true;
+    p3a_config.star_randomness_host = kTestHost;
+
+    ConstellationHelper::RegisterPrefs(local_state.registry());
+
+    url_loader_factory.SetInterceptor(base::BindLambdaForTesting(
+        [&](const network::ResourceRequest& request) {
+          url_loader_factory.ClearResponses();
+
+          std::string response;
+          if (request.url == GURL(std::string(kTestHost) + "/info")) {
+            EXPECT_EQ(request.method, net::HttpRequestHeaders::kGetMethod);
+
+            response = "{\"currentEpoch\":" + base::NumberToString(kTestEpoch) +
+                       ", \"nextEpochTime\": \"" + kTestNextEpochTime + "\"}";
+            info_request_made = true;
+          } else if (request.url ==
+                     GURL(std::string(kTestHost) + "/randomness")) {
+            response = HandleRandomnessRequest(request, kTestEpoch);
+            points_request_made = true;
+          }
+          if (!response.empty()) {
+            if (interceptor_send_bad_response) {
+              response = "this is a bad response that is not json";
+            }
+            url_loader_factory.AddResponse(request.url.spec(), response);
+          }
+        }));
+  }
+
+  void SetUpHelper() {
+    helper = std::make_unique<ConstellationHelper>(
+        &local_state, shared_url_loader_factory,
+        base::BindLambdaForTesting(
+            [this](std::string histogram_name, uint8_t epoch,
+                   std::unique_ptr<std::string> serialized_message) {
+              histogram_name_from_callback = histogram_name;
+              epoch_from_callback = epoch;
+              serialized_message_from_callback = std::move(serialized_message);
+            }),
+        base::BindLambdaForTesting([this](RandomnessServerInfo* server_info) {
+          server_info_from_callback = server_info;
+        }),
+        &p3a_config);
+    task_environment_.RunUntilIdle();
+  }
+
+  content::BrowserTaskEnvironment task_environment_;
+  network::TestURLLoaderFactory url_loader_factory;
```

Comment:
These variables are not named according to convention required by the the style guide [1]. I'm flagging this to raise some awareness, to authors and reviewers, as I've come across a few places today where such cases seemed to have got into master.

I also think we could there are some area for improvement about which members should have been kept private, or protected.

[1] https://google.github.io/styleguide/cppguide.html#Variable_Names